src/edu/csus/ecs/pc2/ui/SubmitSampleRunsPane.java

```diff
@@ -0,0 +1,880 @@
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.logging.Level;
+
+import javax.swing.ButtonGroup;
+import javax.swing.JButton;
+import javax.swing.JCheckBox;
+import javax.swing.JFileChooser;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JRadioButton;
+import javax.swing.JScrollPane;
+import javax.swing.JSplitPane;
+import javax.swing.JTextArea;
+import javax.swing.JTextField;
+import javax.swing.ScrollPaneConstants;
+import javax.swing.SwingConstants;
+import javax.swing.SwingUtilities;
+import javax.swing.border.TitledBorder;
+import javax.swing.filechooser.FileFilter;
+
+import edu.csus.ecs.pc2.VersionInfo;
+import edu.csus.ecs.pc2.core.ClipboardUtilities;
+import edu.csus.ecs.pc2.core.FileUtilities;
+import edu.csus.ecs.pc2.core.IInternalController;
+import edu.csus.ecs.pc2.core.log.Log;
+import edu.csus.ecs.pc2.core.model.ClientSettings;
+import edu.csus.ecs.pc2.core.model.IInternalContest;
+import edu.csus.ecs.pc2.core.model.Language;
+import edu.csus.ecs.pc2.core.model.Problem;
+import edu.csus.ecs.pc2.list.ListUtilities;
+import edu.csus.ecs.pc2.ui.team.QuickSubmitter;
+
+/**
+ * A UI that to submit files found in a CDP.
+ * 
+ * 
+ * @author Douglas A. Lane, PC^2 Team, pc2@ecs.csus.edu
+ */
+public class SubmitSampleRunsPane extends JPanePlugin {
+
+	/**
+	 * ClientSettings key for CDP Path
+	 */
+	private static final String CUSTOM_SUBMIT_SAMPLE_CDP_PATH = "CustomSubmitSampleCDPPath";
+
+	private static final long serialVersionUID = -8862440024499524533L;
+
+	private JTextField cdpTextField;
+
+	private JLabel messageLabel;
+
+	private JCheckBox checkBoxSubmitYesSamples;
+
+	private JCheckBox checkBoxSubmitFailingSamples;
+
+	private QuickSubmitter submitter = new QuickSubmitter();
+
+	private JTextArea textArea;
+
+	private SimpleDateFormat hhMMSSformatter = new SimpleDateFormat("hh:mm:ss");
+
+	private int linenumber = 0;
+	/**
+	 * @wbp.nonvisual location=162,159
+	 */
+	private final ButtonGroup judgingTypesButtonGroup = new ButtonGroup();
+	
+	private final ButtonGroup problemsButtonGroup = new ButtonGroup();
+    
+	private final ButtonGroup languagesButtonGroup = new ButtonGroup();
+    
+	
+	/**
+	 * List of selected solutions names and dirs.
+	 */
+	private SubmissionSolutionList submissionSolutionList = null;
+	
+	/**
+	 * List of selected Problems.
+	 * 
+	 */
+	private List<Problem> selectedProblemList = null;
+	
+    private List<Language> selectedLanguageList = null;
+    
+	private int[] selectedJudgingTypeIndexes;
+	
+	private int[] selectedProblemsIndexes;
+	
+	private int[] selectedLanguagesIndexes;
+	
+	private JLabel selectedNosLabel = new JLabel("None selected");
+	
+	private JRadioButton submitAllJudgingTypesRadioButton = new JRadioButton("Submit All");
+	
+	private JRadioButton submitSelectedJudgingTypeRadioButton = new JRadioButton("Submit Selected");
+	
+    private JCheckBox checkBoxProblems = null;
```

Comment:
It would be nice if you would do a Ctrl-Shift-F on the entire class; there are many places where the formatting/spacing/indenting is inconsistent, which makes it harder for reviewers to follow...

Reply:
Hmmm... when I pull the code into Eclipse, it does not exhibit the confusing indentation noted here (and also several places below).  I'm not sure what's going on; maybe a GitHub anomaly?